accepted/future-releases/unified-collections/feature-specification.md

```diff
@@ -0,0 +1,802 @@
+            `variable` it refers to.
+
+        1.  Evaluate `iterator.current` and bind it to `variable`.
+
+        1.  Evaluate the body element using this procedure in the scope of
+            `variable`.
+
+        1.  If the `for-in` element declares a variable, discard the namespace
+            created for it.
+
+1.  Else, if `element` is an asynchronous `await for-in` element:
+
+    1.  Evaluate the stream expression to a value `stream`. It is a dynamic
+        error if `stream` is not an instance of a class that implements
+        `Stream`.
+
+    1.  Create a new `Future`, `streamDone`.
+
+    1.  Evaluate `await streamDone`.
+
+    1.  Listen to `stream`. On each data event `event` the stream sends:
+
+        1.  If the `for-in` element declares a variable, create a new namespace
+            and a fresh `variable` for it. Otherwise, use the existing
+            `variable` it refers to.
+
+        1.  Bind `event` to `variable`.
+
+        1.  Evaluate the body element using this procedure in the scope of
+            `variable`. If this raises an exception, complete `streamDone` with
+            it as an error.
+
+        1.  If the `for-in` element declares a variable, discard the namespace
+            created for it.
+
+    1.  If `stream` raises an exception, complete `streamDone` with it as an
+        error. Otherwise, when all events in the stream are processed, complete
+        `streamDone` with `null`.
+
+1.  Else, `element` is a C-style `for` element:
+
+    1.  Evaluate the initializer clause of the element, if there is one.
+
+    1.  Loop:
+
+        1.  If the initializer clause declares a variable, create a new
+            namespace and bind that variable in that namespace.
+
+        1.  Evaluate the condition expression to a value `condition`. If there
+            is no condition expression, use `true`.
+
+        1.  If the boolean conversion of `condition` is not `true`, exit the
+            loop.
+
+        1.  Evaluate the body element using this procedure in the namespace of
+            the variable declared by the initializer clause if there is one.
+
+        1.  If there is an increment clause, execute it.
+
+        1.  If the initializer clause declares a variable, discard the namespace
+            created for it.
+
+The procedure theoretically supports a mixture of expressions and map entries,
+but the static semantics prohibit an actual literal containing that. Once the
+`result` series of values or entries is produced from the tree of elements:
+
+### Lists
+
+1.  The result of the literal expression is a fresh instance of a class that
+    implements `List<E>` containing the values in `result`, in order.
+
+### Sets
+
+1.  Create a fresh instance `set` of a class that implements `Set<E>`.
+
+1.  For each `value` in `result`:
+
+    1.  If `set` contains any value `existing` that is equal to `value`
+        according to `existing`'s `==` operator, do nothing.
```

Comment:
except if constant, then it's a compile-time error.

Reply:
Covered up in the constant section.

Reply:
Doesn't actually work that way.
I think you should say it here: 
> 1. if ... `==` operator:
>     * if the literal is constant, it is a compile-time error.
>     * otherwise do nothing.

Same for map keys below.